dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/controller/impl/WorkUnitThreadControllerImpl.java

```diff
@@ -122,6 +133,27 @@ private boolean startWorkUnitIfNotStopping() {
                                                                                              runtimeWorkUnitExecution.getTopLevelInstanceId(),
                                                                                              runtimeWorkUnitExecution.getTopLevelExecutionId() },
                                                    logger);
+
+                // If we're not a split/flow we're a job, so log the original JSL
+                if (!(runtimeWorkUnitExecution instanceof RuntimeSplitFlowExecution)) {
+
+                    JobInstanceEntity instanceEntity = getPersistenceManagerService().getJobInstance(runtimeWorkUnitExecution.getTopLevelInstanceId());
+                    String jsl = instanceEntity.getJobXml();
+                    JoblogUtil.logToJobLogAndTraceOnly(Level.INFO, "display.unresolved.jsl", new Object[] { jsl }, logger);
+                }
+
+                // Print the resolved JSL for the job or flow
+                JSLJob jslJob = jobNavigator.getRootModelElement();
+                ModelSerializer<JSLJob> ms = ModelSerializerFactory.createJobModelSerializer();
+                String xml = ms.serializeModel(jslJob);
+                String prettyXml = formatXML(xml);
+
+                // Set the type.  Partitions are handled elsewhere, step doesn't go through here
+                String type = "job";
+                if (runtimeWorkUnitExecution instanceof RuntimeSplitFlowExecution) {
+                    type = "flow";
+                }
+                JoblogUtil.logToJobLogAndTraceOnly(Level.INFO, "display.resolved.jsl", new Object[] { type, prettyXml }, logger);
             }
```

Comment:
Probably good to log resolution at the flow level even if it's basically the same as at the job level.     I'm curious, is the "root" model element for a flow something further down the chain than the full job?   I think it is, IIRC, and we're overloading the model.

Reply:
It is kind of funky...you get a magic 'job' element and the job level properties, and then just the flow, here's the start (the '1' is the job id, aName is a job property).  
 <job id=":1:split1:flow1">
   <properties>
      <property name="aName" value="Fred"/>
   </properties>
   <flow id="flow1">

Reply:
OK, it's built in **ParallelStepBuilder.buildFlowInSplitSubJob(long, JobContext, Split, Flow)** I see now.